build/jobs/analyze.yml

```diff
@@ -127,4 +129,13 @@ steps:
     GdnBreakGdnToolArmory: ${{ eq(parameters.analyzeARMTemplates, 'true') }}
     GdnBreakGdnToolCredScan: true
     GdnBreakGdnToolCSRF: true
-    GdnBreakGdnToolRoslynAnalyzers: true
\ No newline at end of file
+    GdnBreakGdnToolRoslynAnalyzers: true
+
+- task: TSAUpload@2
+  displayName: 'TSA upload'
+  inputs:
```

Comment:
I believe you only want to create bugs for main branch builds.  That is, if someone is running the build on a private branch, automatically-created bugs is not desired.  See Ram's PRs on how to create a filter main branch.

Reply:
At the task level, something like this: 
condition: and(succeeded(), eq(variables['build.sourceBranch'], 'refs/heads/main'))
  
See https://microsofthealth.visualstudio.com/Health/_git/health-paas/pullrequest/23225?_a=files&path=/build/templates/build.sdl.yml